Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,111 @@
+  public fileprivate(set) var name: String
+
+  public var image: AssetImageTypeAlias {
+    let bundle = Bundle(for: BundleToken.self)
+    let img = image(for: Bundle.main) ?? image(for: bundle)
+    guard let result = img else { fatalError("Unable to load image named \(name).") }
+    return result
+  }
+
+  private func image(for bundle: Bundle) -> AssetImageTypeAlias? {
+    #if os(iOS) || os(tvOS)
+    let image = AssetImageTypeAlias(named: name, in: bundle, compatibleWith: nil)
+    #elseif os(OSX)
+    let image = bundle.image(forResource: NSImage.Name(name))
+    #elseif os(watchOS)
+    let image = AssetImageTypeAlias(named: name)
+    #endif
+    return image
+  }
+}
+```
+
+We should also configure SwiftGen to generate identifiers for additional assets provided by application-specific targets. We can configure `SwiftGen` to name `enum`s like `enum TargetAsset` to not have conflicts with BabylonDependencies' `enum Asset`.
+
+### Other rules
+ 
+Besides standard rules of adding new assets we should also take care when: 
+1. We are updating an icon we should check if any other place uses it. If so and that other place shouldn't be updated we should create a new icon with an updated design. 
+2. If we are removing some code which is using some icons we should check if that code was the last place which was using a given icon, if so icon should be deleted from `Assets.xcassets`. This step could be automated: if we are not using a specific image its identifier won't be used across the project. We can write the script which is checking the usage of every asset and when it is not used anywhere the asset in `Assets.xcassets` the catalog can be deleted. 
+
+## Impact on existing codebase
+ 
+Unfortunately, in our codebase assets are located in different places. If we will agree on the above set of rules, they should be applied for newly created assets. We should also make an effort to eliminate technical debt and migrate all existing assets into `BabylonDependencies`. During that process, we should have in mind it is a good opportunity to improve assets' names for SwiftGen integration. Keeping every asset in one location potentially shouldn’t increase the size of the target application on the condition that the app has all our frameworks linked. 
+
+## Alternatives considered 
+
+1. We could try to systematize the way we include assets in specific feature frameworks but it can cause problems described in the motivation section. 
+
+2. (`@deprecated - This point was discussed at the meeting, because we are not including assets in the design library and we are using SwiftGen abstraction to access assets, usage of @dynamicMemberLookup is not adding us any extra value.`) Instead of accessing icons or images by subscripts `designLibrary.tokens.icons[Asset.close]` we could use a new feature of Swift 5.1 `@dynamicMemberLookup` which could be combined with `KeyPath`. Then we could write just `designLibrary.tokens.icons.close`. 
```

Comment:
>  because we are not including assets in the design library

Is this true? Today the agreed way is in  a Renderer I'd use:
```appearance.tokens.icons.close```
Assuming `appearance` is the designLibrary.